src/groupByKey.ts

```diff
@@ -2,6 +2,41 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
+
+    if (arrayForItems.length === 1) {
+      return arrayForItems[0];
+    }
+
+    return arrayForItems;
+  });
```

Comment:
The `map` function is not used correctly here. You're using it to create a new array `moddedItems`, but then you're not using the returned value. This is a misuse of `map` function, which is intended to transform an array into a new array by applying a function to each item. Instead, you're using it as a loop to filter items and push them into `arrayForItems`. This is both inefficient and confusing.